cmd/abapEnvironmentBuild.go

```diff
@@ -0,0 +1,270 @@
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runAbapEnvironmentBuild(config *abapEnvironmentBuildOptions, telemetryData *telemetry.CustomData, utils abapEnvironmentBuildUtils, cpe *abapEnvironmentBuildCommonPipelineEnvironment) error {
+	values, err := generateValues(config)
+	if err != nil {
+		return errors.Wrap(err, "Generating the values from config failed")
+	}
+	conn := new(abapbuild.Connector)
+	if err := initConnection(conn, config, utils); err != nil {
+		return errors.Wrap(err, "Connector initialization for communication with the ABAP system failed")
+	}
+	finalValues, err := runBuild(conn, config, utils, values)
+	if err != nil {
+		return errors.Wrap(err, "Error during execution of build framework")
+	}
+	cpe.build.values = finalValues
+	return nil
+}
+
+func initConnection(conn *abapbuild.Connector, config *abapEnvironmentBuildOptions, utils abapEnvironmentBuildUtils) error {
+	var connConfig abapbuild.ConnectorConfiguration
+	connConfig.CfAPIEndpoint = config.CfAPIEndpoint
+	connConfig.CfOrg = config.CfOrg
+	connConfig.CfSpace = config.CfSpace
+	connConfig.CfServiceInstance = config.CfServiceInstance
+	connConfig.CfServiceKeyName = config.CfServiceKeyName
+	connConfig.Host = config.Host
+	connConfig.Username = config.Username
+	connConfig.Password = config.Password
+	connConfig.MaxRuntimeInMinutes = config.MaxRuntimeInMinutes
+	connConfig.CertificateNames = config.CertificateNames
+
+	if err := conn.InitBuildFramework(connConfig, utils, utils); err != nil {
+		return err
+	}
+	conn.MaxRuntime = utils.getMaxRuntime()
+	conn.PollingInterval = utils.getPollingIntervall()
+	return nil
+}
+
+// ***********************************Run Build***************************************************************
+func runBuild(conn *abapbuild.Connector, config *abapEnvironmentBuildOptions, utils abapEnvironmentBuildUtils, values abapbuild.Values) (string, error) {
+	build := myBuild{
+		Build: abapbuild.Build{
+			Connector: *conn,
+		},
+		abapEnvironmentBuildOptions: config,
+	}
+
+	if err := build.Start(values); err != nil {
+		return "", err
+	}
+
+	if err := build.Poll(); err != nil {
+		return "", err
+	}
+
+	if err := build.PrintLogs(); err != nil {
+		return "", err
+	}
+	if err := build.EndedWithError(); err != nil {
+		return "", err
+	}
+	if err := build.Download(); err != nil {
+		return "", err
+	}
+	if err := build.Publish(utils); err != nil {
+		return "", err
+	}
+
+	finalValues, err := build.GetFinalValues()
+	if err != nil {
+		return "", err
+	}
+	return finalValues, nil
+}
+
+type myBuild struct {
+	abapbuild.Build
+	*abapEnvironmentBuildOptions
+}
+
+func (b *myBuild) Start(values abapbuild.Values) error {
+	if err := b.Build.Start(b.abapEnvironmentBuildOptions.Phase, values); err != nil {
+		return errors.Wrap(err, "Error starting the build framework")
+	}
+	return nil
+}
+
+func (b *myBuild) EndedWithError() error {
```

Comment:
Function Name -> EvaluateIfEndWithError? EvaluteIfBuildSuccessful?

Reply:
Now renamed to EvaluteIfBuildSuccessful